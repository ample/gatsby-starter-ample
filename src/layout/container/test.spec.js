import React from "react"
import renderer from "react-test-renderer"

import { component as Component, container as Container, fixtures } from "."

describe("Container", () => {
  it("does not cause errors without props", () => {
    const tree = renderer.create(<Container {...fixtures.empty} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders with one component", () => {
    const tree = renderer.create(<Container {...fixtures.default} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("Component", () => {
  it("returns null when no data", () => {
    const tree = renderer.create(<Component />).toJSON()
    expect(tree).toEqual(null)
  })
  it("returns null when no template property", () => {
    const tree = renderer.create(<Component template={null} />).toJSON()
    expect(tree).toEqual(null)
  })
  it("returns null when no component mapping exists", () => {
    const tree = renderer.create(<Component template={"WRONG-COMP-NAME"} />).toJSON()
    expect(tree).toEqual(null)
  })
  it("renders properly with valid props", () => {
    const tree = renderer.create(<Component data={fixtures.default.components[0]} />).toJSON()
    expect(tree).not.toEqual(null)
    expect(tree).toMatchSnapshot()
  })
  it("transforms data (using form as an example)", () => {
    const tree = renderer.create(<Component data={fixtures.form.components[0]} />).toJSON()
    expect(tree).not.toEqual(null)
    expect(tree.children[0].type).toEqual("input")
    expect(tree).toMatchSnapshot()
  })
})
