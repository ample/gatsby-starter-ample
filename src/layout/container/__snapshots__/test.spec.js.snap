// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly with valid props 1`] = `
<a
  className="button"
  href="/"
>
  Hello World
</a>
`;

exports[`Container does not cause errors without props 1`] = `
<div
  className="container"
>
  <div
    className="content"
  />
</div>
`;

exports[`Container renders with one component 1`] = `
<div
  className="container"
>
  <div
    className="content"
  />
</div>
`;
